src/libs/ajax/AzureStorage.ts

```diff
@@ -55,24 +68,30 @@ export const AzureStorage = signal => ({
     const xml = new window.DOMParser().parseFromString(text, 'text/xml')
     const blobs = _.map(
       blob => ({
-        name: _.head(blob.getElementsByTagName('Name')).textContent,
-        lastModified: new Date(
-          _.head(blob.getElementsByTagName('Last-Modified')).textContent
-        ).getTime()
+        name: _.head(blob.getElementsByTagName('Name'))?.textContent,
+        lastModified: _.head(blob.getElementsByTagName('Last-Modified'))?.textContent
       }),
       xml.getElementsByTagName('Blob')
-    )
+    ) as AzureFileRaw[]
 
     const filteredBlobs = _.filter(blob => _.endsWith(suffixFilter, blob.name), blobs)
     return filteredBlobs
   },
 
-  listNotebooks: async workspaceId => {
+  listNotebooks: async (workspaceId: string): Promise<AnalysisFile[]> => {
     const notebooks = await AzureStorage(signal).listFiles(workspaceId, '.ipynb')
-    return _.map(notebook => ({ ...notebook, application: tools.Jupyter.label }), notebooks)
+    return _.map(notebook => ({
+      lastModified: new Date(notebook.lastModified).getTime(),
+      name: notebook.name as AbsolutePath,
+      ext: getExtension(notebook.name),
+      displayName: getDisplayName(notebook.name),
+      fileName: getFileName(notebook.name),
+      tool: toolLabelTypes[tools.Jupyter.label],
+      cloudProvider: cloudProviderTypes.AZURE
+    }), notebooks)
   },
 
-  blob: (workspaceId, blobName) => {
+  blob: (workspaceId: string, blobName: string): any => {
```

Comment:
could we try to use 'unknown' instead of 'any' for cases like this where the type is legitimately unknown?

Reply:
Hmm... this was from an earlier iteration to make the transpiler happy when mocking this contract. I think, in this instance, it's better to omit the return type and let typescript infer.